h/browser/chrome/lib/hypothesis-chrome-extension.js

```diff
@@ -162,20 +187,33 @@ function HypothesisChromeExtension(dependencies) {
     state.clearTab(tabId);
   }
 
+  // installs or uninstalls the sidebar from a tab when the H
+  // state for a tab changes
   function updateTabDocument(tab) {
     // If the tab has not yet finished loading then just quietly return.
     if (tab.status !== TAB_STATUS_COMPLETE) {
       return Promise.resolve();
     }
 
-    if (state.isTabActive(tab.id)) {
-      return sidebar.injectIntoTab(tab).catch(function (err) {
-        tabErrors.setTabError(tab.id, err);
-        state.errorTab(tab.id);
+    var isInstalled = state.getState(tab.id).extensionSidebarInstalled;
+    if (state.isTabActive(tab.id) && !isInstalled) {
+      // optimistically set the state flag indicating that the sidebar
+      // has been installed
+      state.setState(tab.id, {
+        extensionSidebarInstalled: true,
       });
+      return sidebar.injectIntoTab(tab)
+        .catch(function (err) {
+          tabErrors.setTabError(tab.id, err);
+          state.errorTab(tab.id);
+        });
     }
-    else if (state.isTabInactive(tab.id)) {
-      return sidebar.removeFromTab(tab);
+    else if (state.isTabInactive(tab.id) && isInstalled) {
+      return sidebar.removeFromTab(tab).then(function () {
```

Comment:
This is the only place that `removeFromTab()` is called. Conceptually the approach is that when you click the browser action, a page load completes or some other event happens it changes the H state of the tab. This function is called in response to an H tab state change and if the tab is inactive but the sidebar is currently installed then it will be removed.